src/docmanager/index.ts

```diff
@@ -399,7 +442,10 @@ class DocumentManager {
   }
 
   private _data: { [key: string]: Private.IDocumentData } = Object.create(null);
-  private _fileTypes: { [key: string ]: string[] } = Object.create(null);
+  private _modelFactories: { [key: string]: IModelFactory } = Object.create(null);
```

Comment:
I prefer this API, and was bitten in `jupyter-js-services` by module-private `Map` that had not properly instantiated when I wanted to use it.